diesel/src/connection/statement_cache.rs

```diff
@@ -1,12 +1,70 @@
 use std::any::TypeId;
 use std::borrow::Cow;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::hash::Hash;
 
 use backend::Backend;
 use query_builder::*;
 use result::QueryResult;
 
 #[doc(hidden)]
 #[allow(missing_debug_implementations)]
+pub struct StatementCache<DB: Backend, Statement> {
+    pub cache: RefCell<HashMap<StatementCacheKey<DB>, Statement>>,
+}
+
+impl<DB, Statement> StatementCache<DB, Statement> where
+    DB: Backend,
+    DB::TypeMetadata: Clone,
+    DB::QueryBuilder: Default,
+    StatementCacheKey<DB>: Hash + Eq,
+{
+    pub fn new() -> Self {
+        StatementCache {
+            cache: RefCell::new(HashMap::new())
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.cache.borrow().len()
+    }
```

Comment:
Thanks for testing clippy! :trollface: 

```
error: item `connection::statement_cache::StatementCache<DB, Statement>` has a public `len` method but no corresponding `is_empty` method
  --> src/connection/statement_cache.rs:29:5
   |
29 |       pub fn len(&self) -> usize {
   |  _____^ starting here...
30 | |         self.cache.borrow().len()
31 | |     }
   | |_____^ ...ending here
   |
   = note: #[deny(len_without_is_empty)] implied by #[deny(warnings)]
note: lint level defined here
  --> src/lib.rs:7:9
   |
7  | #![deny(warnings, missing_debug_implementations, missing_copy_implementations)]
   |         ^^^^^^^^
   = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#len_without_is_empty
```

(This was okay before as it was `#[cfg(test)]`)